signer/src/storage/postgres.rs

```diff
@@ -1921,13 +1952,94 @@ impl super::DbRead for PgStore {
 
     async fn get_swept_withdrawal_requests(
         &self,
-        _chain_tip: &model::BitcoinBlockHash,
-        _context_window: u16,
+        chain_tip: &model::BitcoinBlockHash,
+        context_window: u16,
     ) -> Result<Vec<model::SweptWithdrawalRequest>, Error> {
-        // TODO: This can use a similar query to
-        // `get_swept_deposit_requests()`, but using withdrawal tables instead
-        // of deposit.
-        unimplemented!()
+        // The following tests define the criteria for this query:
+        // - [X] get_swept_deposit_requests_returns_swept_deposit_requests
+        // - [X] get_swept_deposit_requests_does_not_return_unswept_deposit_requests
+        // - [X] get_swept_deposit_requests_does_not_return_deposit_requests_with_responses
+        // - [X] get_swept_deposit_requests_response_tx_reorged
+
+        let Some(stacks_chain_tip) = self.get_stacks_chain_tip(chain_tip).await? else {
+            return Ok(Vec::new());
+        };
```

Comment:
You should probably take the stacks chain tip as an input into this function.

Reply:
We can keep it as is for now, it's an optimization. We can update them both some time later.